testing/tests/DevExpress.ui.widgets.dataGrid/dataController.tests.js

```diff
@@ -5405,6 +5405,65 @@ QUnit.test("Inserting Row", function(assert) {
     assert.equal(this.dataController.items()[1].dataIndex, 0);
 });
 
+//T521968
+QUnit.test("Inserting several rows for cell editing mode", function(assert) {
+    var array = [
+        { name: 'Alex', phone: '55-55-55' },
+        { name: 'Dan', phone: '98-75-21' }
+    ];
+
+    this.options.editing = {
+        mode: "cell",
+        allowAdding: true
+    };
+
+    var dataSource = createDataSource(array, { key: 'name' });
+
+    this.dataController.setDataSource(dataSource);
+    dataSource.load();
+
+    //act
+    this.addRow();
+    this.addRow();
+
+    //assert
+    assert.equal(this.dataController.items().length, 4, "two rows are added");
```

Comment:
It is better to assign items to a separate variable: 
`var items = this.dataController.items();
assert.equal(items .length, 4, "two rows are added");
.....`